livre/core/pipeline/PortInfo.h

```diff
@@ -0,0 +1,81 @@
+/* Copyright (c) 2011-2016, EPFL/Blue Brain Project
+ *                     Ahmet Bilgili <ahmet.bilgili@epfl.ch>
+ *
+ * This file is part of Livre <https://github.com/BlueBrain/Livre>
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License version 3.0 as published
+ * by the Free Software Foundation.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#ifndef _PortInfo_h_
+#define _PortInfo_h_
+
+#include <livre/core/pipeline/PortType.h>
+
+#include <livre/core/types.h>
+
+namespace livre
+{
+
+/**
+ * PortInfo holds the port information ( name, data type )
+ */
+struct PortInfo final : public PortType
```

Comment:
I suggest this alternative:

    #include <string>
    #include <typeinfo>
    #include <typeindex>
    
    template< typename T >
    struct tag
    {
        typedef T type; // Or using T = type in c++11
    };
    
    
    struct PortInfo
    {
        template< typename T >
        PortInfo( const std::string name_, const tag< T >& )
            : name(name_)
            , type( std::type_index( typeid( T )))
        {
        }
    
        const std::string name;
        const std::type_index type;
    };
    
    
    int main()
    {
        PortInfo a( "foo", tag< int >( ));
    }

Then makePortInfo overloads are not needed at all.